### PR TITLE
working fall-through algorithm

### PR DESCRIPTION
**Pull request by jonathanbratt**
it's not pretty, and it's slow.

### CODE REVIEW

File: R/morphemepiece.R [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                                    original_start = TRUE,
+                                    original_end = TRUE,
+                                    prefixes) {
+    if (word == "") {
+        return(word)
+    }
+    
+    # deduce whether this is a valid_start from the ##
+    valid_start <- FALSE
+    if (stringr::str_starts(word, pattern = "##", negate = TRUE)) {
```
Comment (jonthegeek) [inline R/morphemepiece.R:66]: Simplify:
`valid_start <- stringr::str_starts(word, pattern = "##", negate = TRUE)`

**Review by jonthegeek — CHANGES_REQUESTED** [review]
I'll approve this with a couple tweaks, but I definitely think we should try a couple more iterations of ways to do this. It feels like it HAS to be possible to (drastically) simplify.

File: R/morphemepiece.R (hunk 1 above)
Comment (jonathanbratt) [reply R/morphemepiece.R:66]: good call!

File: R/morphemepiece.R [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+#' is intended to be used as a "fall-back" for words that are not found in the
+#' morphemepiece lookup table.
+#'
+#' @param word Character; word to tokenize.
+#' @param vocab A morphemepiece vocabulary.
+#'
+#' @return A character vector corresponding to the original word, broken into 
+#' pieces.
+#'
+#' @keywords internal
```
[the remark this answers is not in the document]
Comment (jonathanbratt) [reply R/morphemepiece.R:151]: exporting a wrapping function, as per conversation.

File: R/morphemepiece.R [hunk 3]
```diff
@@ -23,7 +23,6 @@
 #' @importFrom magrittr %>%
 #' @name %>%
 #' @rdname pipe
-#' @export
 #' @param lhs,rhs A vector of fields or a tibble of fields and values, and a
 #'   function to apply to them.
 NULL
```
Comment (jonthegeek) [inline R/morphemepiece.R]: Let's start using the more-correct import methodology:

```
#' @importFrom magrittr %>%
magrittr::`%>%`
```

File: R/morphemepiece.R [hunk 4]
```diff
@@ -62,17 +61,14 @@ NULL
...
-    }
+    valid_start <- stringr::str_starts(word, pattern = "##", negate = TRUE)
     
     word <- stringr::str_remove_all(word, pattern = "#")
     
     match <- stringr::str_locate(string = word, pattern = mp_vocab_nohash) %>%
         tibble::as_tibble() %>%
         # need an index...
-        dplyr::mutate(which_wp = row_number()) %>%
+        dplyr::mutate(which_wp = dplyr::row_number()) %>%
```
Comment (jonthegeek) [inline R/morphemepiece.R:62]: Good catch!

**Review by jonthegeek — APPROVED** [review]
I can't undo my comment about the pipe, but obviously that's fine now (I think it MIGHT have gone away since the code it was attached to is gone now). Obviously this will need work, but woot to having a thing!

**Pull request merged**